### ISSUE TITLE
Raspberry Pi - image capture, select time out

### ISSUE THREAD

**Issue by jonhare** [openimaj/openimaj#48]
OP-49: Hi :)

Just wanted to share newest findings. Hepin recently compiled OpenIMAJGrabber.so on his Raspberry Pi and found very interesting problem. After some time, single image capture takes more and more time, until user receive "select timeout" error from [capture.cpp:165|http://sourceforge.net/p/openimaj/code/1811/tree/trunk/hardware/core-video-capture/src-native/linux/capture.cpp#l165]. 

For more details please see issue [webcam-capture#42|https://github.com/sarxos/webcam-capture/issues/42]. 

Do you maybe know what can be a root cause?

**Comment by jonhare** [comment]
Original reporter: @sarxos

**Comment by jonhare** [comment]
jonhare:  Will look into this asap

**Comment by jonhare** [comment]
jonhare:  I think this might be a hardware or driver problem. I've had the latest build of the native libraries running for about three hours (ARMEL Raspbian) and it's still happy; I'll keep it running overnight to see if it breaks.

**Comment by jonhare** [comment]
sarxos:  Hi,

Thank you for looking into this Jon. 

This issue has been initially reported by Hepin - he supposed it can be caused by some powering issues, but I don't have any details on his findings.

**Comment by jonhare** [comment]
hepin1989:  Thanks for your both ,this really may be the low power issue ,or the hardware issue.
I found that some times i close the webcam ,or set to an different resolution,then will cause the select time out some times .but this may be fixed if I disconnect the webcam and reconnect it to the Raspberry pi ,it works.
Another issue is that ,on the raspberry pi ,it may run and then the fps going down ,finally the select time out comes out,in my cause ,it just cost about 20 minutes to detect this issue .
Sorry for my bad English,thanks for your both guys great work!

**Comment by jonhare** [comment]
jonhare:  OK, mine was still running happily this morning; we'll assume it's a power or driver problem.

**Comment by jonhare** [comment]
hepin1989:  @Jonathon,OK,Could you tell me how is the FPS on your raspberry pi?
On mine ,the FPS is so low,about ~5,and the raspberry team will release the camera model,but it was handled by openomx IL,and does not support v4l2 now,
so that means we can't get the image via Bartosz's webcam capture lib ?
So ,cloud we just read the data from the GPU?
thanks,I will going to read thinking in C++ and start to learn some C++ key points
```
It should be mentioned that the currently software interface is not integrated into any standard linux system…

i.e. no OpenCV or V4L2 drivers

Basically its more like the OpenMaxIL interface of the video (although we use something we know as MMAL the MultiMedia Adaptation Layer) so for those of you interested in doing something using OpenCV then that’s not going to fly…

Whereas if someone knew V4L2 drivers and knew our OpenMaxIL / MMAL interface and would like to make something then that would be interesting…

Gordon
```
